### PR TITLE
Add way of detecting the physical keyboard layout

### PR DESCRIPTION
**Pull request by z3ntu**
- [x] Driver support
- [x] Daemon suport (methods)
- [x] Daemon support (devices)
- [x] Pylib support
- [x] Fake driver support

~~I'd like to make point nr3 based on #463~~ done

### CODE REVIEW

**Comment by z3ntu** [comment]
image

**Comment by z3ntu** [comment]
When this is merged, update the 'Keyboard layouts' wiki page with the following content:



You can get the physical keyboard layout from your keyboard with the command `cat /sys/bus/hid/drivers/razerkbd/0*/kbd_layout` which outputs a 2-digit number.

Here's a list of known id-name mappings.

| number | layout |
| ------ | ------ |
|   01   | en_US |
|   03   | de_DE |
|   06   | en_GB |

If your keyboard has another number than those, please open an issue with the 'number' and your keyboard layout (preferably including a screenshot).

File: pylib/openrazer/client/devices/__init__.py [hunk 1]
```diff
@@ -31,6 +31,10 @@ def __init__(self, serial, vid_pid=None, daemon_dbus=None):
         self._type = str(self._dbus_interfaces['device'].getDeviceType())
         self._fw = str(self._dbus_interfaces['device'].getFirmware())
         self._drv_version = str(self._dbus_interfaces['device'].getDriverVersion())
+        if self._has_feature('razer.device.misc', 'getKeyboardLayout'):
```
Comment (z3ntu) [inline pylib/openrazer/client/devices/__init__.py]: TODO: Make that a proper capability (`"kbd_layout"`)

**Review by lah7 — APPROVED** [review]
Great job! Installed on Ubuntu, rebooted and my keyboard is indeed `en_GB`. The fake driver works too (`en_US`).

**Comment by terricain** [comment]
Was too lazy to look at the code but:
* What does that DBUS endpoint return if its not recognised?
* Also is the driver outputting hex, and if so are you accounting for it

Also might not be a bad shout to have the GUIs to prompt the user to make a github issue (premade one if thats dooable) with unknown values

**Comment by z3ntu** [comment]
I am actually not sure what the driver is outputting 😄 on my keyboard it outputs `03` (+newline). But it doesn't matter too much as I'm comparing the string and not converting it into anything.
Unknown values are handled and the daemon converts them to `unknown`.

**Pull request merged**